src/Usb.Net/Windows/WindowsUsbDeviceFactory.cs

```diff
@@ -41,7 +42,22 @@ public IDevice GetDevice(ConnectedDeviceDefinition deviceDefinition)
         #region Private Static Methods
         protected override ConnectedDeviceDefinition GetDeviceDefinition(string deviceId)
         {
-            return GetDeviceDefinitionFromWindowsDeviceId(deviceId, DeviceType.Usb, Logger);
+            // Getting device interface handle.
```

Comment:
I think I understand what you are doing here.

Are you saying that we need to open a handle for each device to get the details you need?

Reply:
yes. previously your code doesnt even call the function to get the details i need so im trying to figure out how to get handle (i dont know if it is right) and call that function.

Reply:
Understood

Reply:
is it right though to create handle like that?

Reply:
I can't answer without doing research, but probably.

Reply:
do you have any doc abt `USB_DEVICE_DESCRIPTOR` buffer?